src/Transform/ONNX/ConstProp.cpp

```diff
@@ -843,6 +848,27 @@ Value ConstPropReshape(
       .getResult();
 }
 
+//===----------------------------------------------------------------------===//
+// Code to perform constant propagation for ConstantOfShape.
+//===----------------------------------------------------------------------===//
+
+Value ConstPropConstantOfShape(PatternRewriter &rewriter, Value replacingValue,
+    Value shape, Attribute value) {
+  ConstPropCounters::count("ConstantOfShape", {shape});
+  llvm::SmallVector<int64_t, 4> shapeVector;
+  ElementsAttr shapeAttr =
+      getONNXConstantOp(shape).getValueAttr().cast<ElementsAttr>();
+  for (mlir::IntegerAttr dim : shapeAttr.getValues<IntegerAttr>())
+    shapeVector.push_back(dim.getInt());
+
+  ElementsAttr constElements = value.cast<ElementsAttr>();
+  OnnxElementsAttrBuilder elementsBuilder(rewriter.getContext());
+  ElementsAttr expandedElements =
+      elementsBuilder.expand(constElements, shapeVector);
```

Comment:
if shapeVector is empty (the result is a scalar) then this stretches the functionality of expand because we "contract" constElements's 1D shape to 0D

looking at the implementation of expand() it's possible that it could work but can you please add a lit test for this? and maybe it's best to special case this with something like:
```
if (shapeVector.empty()) expandedElements = elementsBuilder.reshape(constElements, shapeVector);
```